client/reader/recommendations/posts/index.jsx

```diff
@@ -0,0 +1,26 @@
+var React = require( 'react' );
```

Comment:
The main point about the guidelines is that sub-folders are not taken into account, and any component is considered to coexist under a flat hierarchy. (That gives us the flexibility that any folder is completely autonomous and we can group them as needed.) In this case I'd just place `posts-recommendations` in `/reader/`'s root or make posts-recommendations a sub-component of `reader/recommendations` if it's not meant to be used elsewhere at all.

Not caring about folder structure makes it really straightforward how a class should be named — it only has to look at its folder name.